designs/ClientDevProcess.md

```diff
@@ -0,0 +1,145 @@
+
+Another way to think about Epics is: if there is a customer-facing feature, an Epic should be the parent of all tasks from spec'ing to writing the actual code to test to insertion to shipping and writing docs.
+
+Concretely, these are GitHub issues tagged as Epic, and they navigate the board slightly different from regular issues:
+
+1. When any issue under an Epic is In Progress, the associated Epic should be moved to the In Progress pipeline as well.
+2. Epics should remain under "In Review" or "Validating" until the related change is available to customers. That is, they shouldn't be closed until released.
+
+Tracking Epics in this way allows customers and stakeholders to see a more fine-grained view of the status of any KR work by going to the Board and [filtering by Epics and hiding subtasks](https://github.com/NuGet/Home#workspaces/nuget-client-team-55aec9a240305cf007585881/board?epics:settings=epicsOnly&filterLogic=any&repos=29996513).
+
+## Sprint Tracking
+
+Previously, the Client Team used GitHub labels to track its 3-week sprints, using the format `Sprint <number>`, and Milestones to track release versions. Now, the Client Team uses GitHub Milestones to track sprint tasks. This is done for one main reason: enabling the [Velocity Tracking](https://github.com/NuGet/Home#workspaces/nuget-client-team-55aec9a240305cf007585881/reports/velocity) and [Burndown](https://github.com/NuGet/Home#workspaces/nuget-client-team-55aec9a240305cf007585881/reports/burndown?milestoneId=4983201) reports.
+
+Issues start off without an assigned sprint. During Sprint Planning at the beginning of each Sprint, items in the Backlog should be assigned to an appropriate Sprint Milestone.
+
+Over time, doing things this way will allow the Client Team to get a sense of its overall Velocity, which will help with future planning and resourcing.
+
+## Release Tracking
+
+As mentioned above, the Client Team used GitHub Milestones to track releases. Since those are now being used for Sprints (for the sake of ZenHub tracking reports), release tracking is moved to the [Release feature](https://github.com/NuGet/Home#workspaces/nuget-client-team-55aec9a240305cf007585881/reports/release?release=5e0e5fbd021f7aa0ec95db18) in ZenHub.
+
+When issues are scheduled, they should be assigned an appropriate Release, along with a Sprint.
+
+In order to generate release changelogs, the team should use [the ZenHub API](https://github.com/ZenHubIO/API#get-all-the-issues-for-a-release-report) -- possibly through [ZenHub.NET](https://github.com/AlexGhiondea/ZenHub.NET/blob/21cd562b30570594beb3842b28c372d66f87dcc6/src/ZenHubReleaseClient.cs#L84-L90).

```

Comment:
We also discussed having a Changelog.md that summarizes the changes between releases. I think having an explicit file in GitHub makes sense considering we should not expect all our customers to use ZenHub.

Reply:
Yes, this would be great, although we should probably have this as a separate proposal. This proposal just makes sure the data is available and that other processes we have keep working.